src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2015 asotobu.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.plaincredentials;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.fileupload.FileItemHeaders;
+import org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+
+public class FileCredentialsTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    @Issue("JENKINS-30926")
+    public void shouldThrowAnExceptionIfFileNameIsBlank() throws IOException {
+        FileCredentials fileCredentials = new FileCredentialsImpl(CredentialsScope.GLOBAL, "1", "",
+                new StubFileItem(), "", "");
```

Comment:
BTW the test is not terribly helpful here because we are not really sure where the blank name came from to begin with. A browser bug? A problem in `config.jelly`?